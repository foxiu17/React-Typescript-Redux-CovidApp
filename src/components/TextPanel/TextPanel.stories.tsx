import React from 'react';

import TextPanel from "./TextPanel";

export default {
  component: TextPanel,
  title: "TextPanel"
};

export const withoutProps = () => <TextPanel />;

export const greenWithProps = () => <TextPanel color="#2ae2ce" title="Total death" count={1000} />;

export const blueWithProps = () => <TextPanel color="#2232ce" title="Total death" count={1000} />;

export const withNotAllProps = () => <TextPanel title="Test title Storybook" />;