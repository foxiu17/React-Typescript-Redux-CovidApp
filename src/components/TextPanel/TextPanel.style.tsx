import styled from 'styled-components';

import { IGridProps } from "../../Interfaces/Styles/StylesInterfaces";

import { Grid as commonGrid } from '../../Assets/Common/Containers.style';

export const Grid = styled(commonGrid)<IGridProps>`
  width: 100%;
  height: 100%;
`;