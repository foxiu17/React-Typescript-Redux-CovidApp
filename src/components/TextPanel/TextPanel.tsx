import React from "react";

import { Grid } from './TextPanel.style';

import { ITextPanelProps } from "../../Interfaces/Interfaces";

const TextPanel = ({ color, title, count }: ITextPanelProps) => {
  return (
    <Grid container justify="center" color={color}>
      <Grid item xs={12}>{title}</Grid>
      <Grid item xs={12}>{count}</Grid>
    </Grid>
  )
};

export default TextPanel;