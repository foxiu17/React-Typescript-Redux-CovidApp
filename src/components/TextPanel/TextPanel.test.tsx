import React from 'react';
import { shallow, mount, ReactWrapper, ShallowWrapper } from "enzyme";
import TextPanel from "./TextPanel";

describe('Test TextPanel component', () => {
  describe("Render tests", () => {
    test('Render component without crashing - without props', () => {
      const component: ShallowWrapper = shallow(<TextPanel />);
      expect(component.exists()).toBe(true);
      expect(component).toMatchSnapshot();
    });

    test('Render component without crashing - with props', () => {
      const component: ShallowWrapper = shallow(<TextPanel color="#2ae2ce" title="Total death" count={1000} />);
      expect(component.exists()).toBe(true);
      expect(component).toMatchSnapshot();
    });
  });
});