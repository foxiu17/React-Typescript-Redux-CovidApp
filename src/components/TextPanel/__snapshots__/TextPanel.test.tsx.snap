// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TextPanel component Render tests Render component without crashing - with props 1`] = `ShallowWrapper {}`;

exports[`Test TextPanel component Render tests Render component without crashing - without props 1`] = `ShallowWrapper {}`;
